src/fsharp/LexFilter.fs

```diff
@@ -427,12 +427,18 @@ type TokenTupPool() =
     [<Literal>]
     let maxSize = 100
 
-    let stack = System.Collections.Generic.Stack(Array.init maxSize (fun _ -> TokenTup(Unchecked.defaultof<_>, Unchecked.defaultof<_>, Unchecked.defaultof<_>)))
+    let mutable currentPoolSize = 0
+    let stack = System.Collections.Generic.Stack(10)
```

Comment:
Why 10 instead of 100 like before?

Reply:
Arbitrary, we will definitely be having more than one for sure in the stack.